node/src/components/block_synchronizer/block_acquisition.rs

```diff
@@ -108,8 +108,8 @@ pub(super) enum BlockAcquisitionState {
     HaveApprovalsHashes(Box<Block>, SignatureAcquisition, DeployAcquisition),
     HaveAllDeploys(Box<Block>, SignatureAcquisition),
     HaveStrictFinalitySignatures(Box<Block>, SignatureAcquisition),
-    HaveFinalizedBlock(BlockHash, Box<FinalizedBlock>, Vec<Deploy>, bool),
-    Complete(BlockHash, u64),
+    HaveFinalizedBlock(Box<Block>, Box<FinalizedBlock>, Vec<Deploy>, bool),
```

Comment:
I have a question here: is `HaveFinalizedBlock` involved in the historical flow? My understanding was that for historical blocks, we go directly from `HaveStrictFinalitySignatures` to `Complete`. The confusion comes from the fact that we're fixing the historical sync with this PR.

So, is this `Box<Block>` only needed to carry on the information about block when we're constructing `Complete` in the forward flow?

Also, this is now super-confusing to have both `Block` and `FinalizedBlock` alongside each other, indicating that the overall design of the block acquisition could use some improvement in the future.

For now, can we at least add a comment of what these two mean in the context of `HaveFinalizedBlock`?

Reply:
We need to keep the block around in `HaveFinalizedBlock` because we'll need it in `Complete` when we want to make the transition. Yes, `HaveFinalizedBlock` is used only in the forward flow, but it's necessary as `Complete` is the final state in both historical and forward sync.

I agree it's confusing to have both the `Block` and the `FinalizedBlock` in there, but in my mind it's not the worst. Some of the things in the state are necessary for the current step (like `FinalizedBlock`), others are needed for future states. It's unfortunate that we have both a `Block` and a `FinalizedBlock`, but we need both it seems. I agree with you though, a refactor would be welcome indeed.

I will add the comment in a later commit.

Reply:
Done.